### DISCUSSION TITLE
saving output to a file while excluding lines containing [download]

### DISCUSSION THREAD

**Discussion by 111100001** [bibanon/tubeup#322] [Q&A]
i have an sh file of multiple tubeup commands the output of which i want saved in a text file EXCLUDING the repetitive [download] progress lines. this command that people said it works on stackoverflow does not work for me. pls i need help

`sh VideosUntilMarch-06-2024.sh | tee >(grep -v '^\[download\]' > UnwantedResults.txt) > resultsWanted.txt`

this command for some reason saves all the output and never filters it. what i noticed that it filters what the console outputs, so the [download] lines dont show up on the console but everything gets saved in the text file resultsWanted.txt unfiltered

**Comment by 111100001** [comment]
FIXED

I took a guess that i should add --line-buffered to the grep command and it worked!

**Comment by vxbinaca** [comment]
use yt-dlp to process your wanted videoIDs, then have Tubeup grab the result one by one
